lib/streamlit/elements/arrow.py

```diff
@@ -145,6 +154,21 @@ def dg(self) -> "DeltaGenerator":
         return cast("DeltaGenerator", self)
 
 
+def show_warning_if_data_is_capped(data: Data = None) -> Data:
```

Comment:
Ahh btw, I think it might be sufficient to actually do the error message handling directly in the `convert_anything_to_df`. In that case, we don't have to call this in all the individual commands. Or was there an issue with executing `st.caption` in `convert_anything_to_df`?

Reply:
The thing is that `convert_anything_to_df` is anyways called in all our components having `dataframe` as input. So, the `caption` would be sent at the right time

Reply:
I was worried that it will be called too many times, but seems like it's not the case 👍 Good call out, it's updated and it's much cleaner 👍